src/qcrypto/simbasics.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import dataclasses
+
+@dataclasses.dataclass
+class Qubit:
+    """
+    Data class which represents the qubit and its quantum state.
+    Quantum state is given by:
+    |psi> = cos(theta/2) |0> + e^{i*phi} * sin(theta/2) |1>
+    """
+
+    def __init__(self, theta, phi, base):
+        self.base = base
+        self.state = np.array([
+            np.cos(theta/2),
+            np.exp(1j * phi) * np.sin(theta/2)
+        ])
+
+    def __str__(self):
+        return "{} |0> + {} |1>".format(self.state[0], self.state[1])
+    
+    def __repr__(self):
+        return "{} |0> + {} |1>".format(self.state[0], self.state[1])
+
+    def normalize(self):
+        """
+        Nomalizes the qubit's quantum state
+        """
+        self.state = self.state / np.sqrt(np.dot(np.conjugate(self.state), self.state))
```

Comment:
Contemplating whether this function should have a `return` statement?

Reply:
Since all this method does is modify the data stored in the class it belongs to, I thought it was appropriate for it not to have a `return` statement. Alternatively, it could be done in a similar way as dataframes, where you can have methods be inplace or not, and in the latter case it would make sense to have a return statement.

Reply:
It doesn't hurt to add it, so I did. We can remove it if we change our mind later on.